### PR TITLE
Create 'Securing application logging' pattern

### LINKED ISSUE #226: Create "Application logging - securing data" pattern
Our engineering patterns are example solutions for addressing problems, meeting standards and living our engineering principles. They should make it easier for engineers to do their jobs. We don't have a standard way of writing patterns as we like to be a bit more flexible with their structure. Please provide us with some information on what you are proposing below

**What is the pattern you are suggesting?**
Transfer of "Application Logging: Avoiding security incidents"

**What would be the benefit to the Home Office of following this pattern**
This could be the mitigation of risks, or the improvement of other aspects, for example quality or speed of delivery

**How do people implement this pattern**
Here you can point to frameworks, third party guidance or anything else that best describes the solution you are proposing

**Additional information**
Add any other information you think would be useful here

- [x] I have looked through our existing patterns and think this is not covered elsewhere
- [x] I have checked through the open issues on the repository and this pattern has not already been suggested

### PR DESCRIPTION
**Pull request by edhamiltonHO**
Will close #226 

# Content change 
I can confirm:
- [x] Content does not include any code or configuration changes (excluding frontmatter information)
- [x] Content meets the content standards
e.g. [Writing a principle](https://ho-cto.github.io/engineering-guidance-and-standards/docs/standards/writing-a-principle/) and [Writing a standard](https://ho-cto.github.io/engineering-guidance-and-standards/docs/standards/writing-a-standard/)
- [x] Content is suitable to open source, i.e.:
    - Content does not relate to unreleased gov policy
    - Content does not refer to anti-fraud mechanisms
    - Content does not include sensitive business logic
- [x] Last updated date for content is correct

### CODE REVIEW

File: docs/patterns/securing-application-logging.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+- limited to the application components that are under investigation (e.g. to investigate a web issue, enable HTTP payload logging but not Kafka messages logging)
+
+---
+
+## Considerations
+
+### Consider whether logging statements are necessary
+Sometimes logging is used for fast-feedback loops when prototyping application functionality, but eventually become redundant, or the application they provide does not provide additional operational or debugging value for an operator searching through logs.
+
+Consider removing logging statements that do not produce value. If the logging statement is added to increase confidence in the application's correctness, consider replacing it with a unit test instead.
```
Comment (robertdeniszczyc2) [inline docs/patterns/securing-application-logging.md:65]: This is almost repeated on line 69

Comment (edhamiltonHO) [reply docs/patterns/securing-application-logging.md]: Have fixed to remove the duplication and bring the two considerations together

File: docs/patterns/securing-application-logging.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+title: Securing application logging
+date: 2023-08-11
+tags: 
+  - Logging
+  - Observability
+  - Secure development
+  - SRE
+  - Security
+  - Software design   
+---
```
Comment (keithkennedyHO) [inline docs/patterns/securing-application-logging.md:13]:
```suggestion
related:
  sections:
    - title: Related links
      items:
        - text: Managing secrets
          href: /standards/managing-secrets/
---
```

We can add our related link here in addition to the content of the page.

File: docs/patterns/securing-application-logging.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+It is common to log user information from an application to assist operations or debugging. Some middleware might be configured to log full configuration, data, or request/response objects, which may contain Personally Identifiable Information (PII), authentication tokens or application secrets.
+
+Logging any of these can make information available to people that have access to log aggregation platforms, or log storage, but are not legally allowed to view PII, or should not otherwise have access to that information. User authentication tokens and application secrets can be used to impersonate users and elevate privilege.
+
+[Sensitive Data Exposure](https://owasp.org/www-project-top-ten/2017/A3_2017-Sensitive_Data_Exposure) has been a recognised risk to secure application development historically. And our [managing secrets]({{ '/standards/managing-secrets/' | url }}) standard has requirements on ensuring logs do not contain tokens or other secrets.
```
Comment (keithkennedyHO) [inline docs/patterns/securing-application-logging.md:19]:
```suggestion
[Sensitive Data Exposure](https://owasp.org/www-project-top-ten/2017/A3_2017-Sensitive_Data_Exposure) has been a recognised risk to secure application development historically. Our [managing secrets]({{ '/standards/managing-secrets/' | url }}) standard has requirements on ensuring logs do not contain tokens or other secrets.
```

**Review by keithkennedyHO — APPROVED** [review]

**Review by robertdeniszczyc2 — APPROVED** [review]

**Pull request merged**